From f3a1c8e8702937f1503da3227b4327e0fd4355a7 Mon Sep 17 00:00:00 2001
From: Hendrik Hagendorn <finnq@lineageos.org>
Date: Sun, 15 Jan 2017 08:45:51 +0100
Subject: [PATCH 5/5] lineagesdk: Refactor battery icon options

* Use the icon blacklist for hiding
* Combine options for hiding status bar icons in system tuner settings

Change-Id: Ie41d71c774a34abe225e2c0a6a0a9fd4316189cd
---
 host/migration/src/LineageSettings.java       |  6 +--
 .../LineageDatabaseHelper.java                | 39 ++++++++++++++++++-
 .../lineageos/providers/LineageSettings.java  |  8 ++--
 3 files changed, 43 insertions(+), 10 deletions(-)

diff --git a/host/migration/src/LineageSettings.java b/host/migration/src/LineageSettings.java
index dabe834..1354096 100644
--- a/host/migration/src/LineageSettings.java
+++ b/host/migration/src/LineageSettings.java
@@ -80,10 +80,8 @@ public final class LineageSettings {
         /**
          * Display style of the status bar battery information
          * 0: Display the battery an icon in portrait mode
-         * 2: Display the battery as a circle
-         * 4: Hide the battery status information
-         * 5: Display the battery an icon in landscape mode
-         * 6: Display the battery as plain text
+         * 1: Display the battery as a circle
+         * 2: Display the battery as plain text
          * default: 0
          * @hide
          */
diff --git a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
index b456327..670b402 100644
--- a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
+++ b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
@@ -51,7 +51,7 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
     private static final boolean LOCAL_LOGV = false;
 
     private static final String DATABASE_NAME = "lineagesettings.db";
-    private static final int DATABASE_VERSION = 11;
+    private static final int DATABASE_VERSION = 12;
 
     private static final String DATABASE_NAME_OLD = "cmsettings.db";
 
@@ -392,6 +392,43 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
             }
             upgradeVersion = 11;
         }
+
+        if (upgradeVersion < 12) {
+            if (mUserHandle == UserHandle.USER_OWNER) {
+                db.beginTransaction();
+                SQLiteStatement stmt = null;
+                try {
+                    stmt = db.compileStatement("SELECT value FROM system WHERE name=?");
+                    stmt.bindString(1, LineageSettings.System.STATUS_BAR_BATTERY_STYLE);
+                    long value = stmt.simpleQueryForLong();
+
+                    long newValue = 0;
+                    switch ((int) value) {
+                        case 2:
+                            newValue = 1;
+                            break;
+                        case 5:
+                            newValue = 0;
+                            break;
+                        case 6:
+                            newValue = 2;
+                            break;
+                    }
+
+                    stmt = db.compileStatement("UPDATE system SET value=? WHERE name=?");
+                    stmt.bindLong(1, newValue);
+                    stmt.bindString(2, LineageSettings.System.STATUS_BAR_BATTERY_STYLE);
+                    stmt.execute();
+                    db.setTransactionSuccessful();
+                } catch (SQLiteDoneException ex) {
+                    // LineageSettings.System.STATUS_BAR_BATTERY_STYLE is not set
+                } finally {
+                    if (stmt != null) stmt.close();
+                    db.endTransaction();
+                }
+            }
+            upgradeVersion = 12;
+        }
         // *** Remember to update DATABASE_VERSION above!
     }
 
diff --git a/sdk/src/java/lineageos/providers/LineageSettings.java b/sdk/src/java/lineageos/providers/LineageSettings.java
index bb97727..b9b2f6d 100644
--- a/sdk/src/java/lineageos/providers/LineageSettings.java
+++ b/sdk/src/java/lineageos/providers/LineageSettings.java
@@ -923,17 +923,15 @@ public final class LineageSettings {
         /**
          * Display style of the status bar battery information
          * 0: Display the battery an icon in portrait mode
-         * 2: Display the battery as a circle
-         * 4: Hide the battery status information
-         * 5: Display the battery an icon in landscape mode
-         * 6: Display the battery as plain text
+         * 1: Display the battery as a circle
+         * 2: Display the battery as plain text
          * default: 0
          */
         public static final String STATUS_BAR_BATTERY_STYLE = "status_bar_battery_style";
 
         /** @hide */
         public static final Validator STATUS_BAR_BATTERY_STYLE_VALIDATOR =
-                new DiscreteValueValidator(new String[] {"0", "2", "4", "5", "6"});
+                new InclusiveIntegerRangeValidator(0, 2);
 
         /**
          * Status bar battery %
-- 
2.17.1

